src/coprocessor/codec/mysql/time.rs

```diff
@@ -128,6 +128,24 @@ impl Time {
         self.tp
     }
 
+    pub fn set_tp(&mut self, tp: u8) -> Result<()> {
+        if self.tp != tp && tp == types::DATE {
```

Comment:
How about convert timestamp to datetime? For example, a timestamp is `2017/08/20 12:00:00 UTC`, and it needs to be converted to datetime in +08:00, what's the expected result?